src/spikeinterface/core/baserecordingsnippets.py

```diff
@@ -225,6 +234,27 @@ def set_probes(self, probe_or_probegroup, group_mode="by_probe", in_place=False)
 
         return sub_recording
 
+    def create_with_probes(self, probe_or_probegroup, group_mode="by_probe"):
```

Comment:
I wonder if it should be `create_with_probe` vs `probes`.

Reply:
Ther are two functions, the one with the plural name is the most general as it takes both probe a list or a probe group, so that's where I want the the warnin to be.  I wish there was only one function and if you see the diff in the changes it seems that that was the intention of one point.

Simplifying probe setting could be a issue of its own, I am not making any changes in that direction in this PR. It is just to get the ball rolling to make the behaivor of the function consistent with setter semantics.